src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -28,6 +29,11 @@ public class AddressBookParser {
      */
     private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
 
+    /**
+     * Used for tag commands.
+     */
+    private final TagCommandParser tagCommandParser = new TagCommandParser();
```

Comment:
Is there a need for the `TagCommandParser` to be initialised here when it might not be used? Other parsers like the AddCommandParser are initialised inside their respective case blocks

Reply:
Yes. I made a mistake. Thanks for pointing it out.